pkg/operators/filter/filter.go

```diff
@@ -81,11 +83,22 @@ func (f *filterOperator) InstanceParams() api.Params {
 	}}
 }
 
+func keyForDataSource(datasources map[string]datasource.DataSource, ds datasource.DataSource) (string, string) {
```

Comment:
```suggestion
func keyForDataSource(ds datasource.DataSource, hasMultiple bool) (string, string) {
```

And then use `len(datasources) > 1` on the caller?